tests/queries/test_contains.py

```diff
@@ -0,0 +1,49 @@
+from django.test import TestCase
+
+from .models import ChildObjectA, ObjectA, ObjectB, ProxyObjectA
+
+
+class ContainsTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.existing = ObjectA.objects.create(name='one')
+        cls.not_saved = ObjectA(name='two')
+        cls.wrong_model = ObjectB.objects.create(name='three', objecta=cls.existing, num=1)
+        cls.proxy_model_existing = ProxyObjectA.objects.create(name='four')
+
+    def test_non_model(self):
+        self.assertRaises(ValueError, ObjectA.objects.contains, object())
+
+    def test_values_querysets(self):
+        """ Make sure .values() and .values_list() QuerySets raises ValueError """
+        with self.assertRaises(ValueError):
+            ObjectA.objects.values_list('pk').contains(self.existing)
+        with self.assertRaises(ValueError):
+            ObjectA.objects.values('pk').contains(self.existing)
+
+    def test_db_queries(self):
+        # DB query tests
+        self.assertIs(ObjectA.objects.contains(self.existing), True)
+        self.assertIs(ObjectA.objects.contains(self.proxy_model_existing), True)
+        self.assertIs(ProxyObjectA.objects.contains(self.existing), True)
+        self.assertIs(ObjectA.objects.contains(self.wrong_model), False)
+        self.assertIs(ObjectA.objects.contains(self.not_saved), False)
+        self.assertIs(ChildObjectA.objects.contains(self.existing), False)
+        self.assertNumQueries(1, ObjectA.objects.contains, self.existing)
+        self.assertNumQueries(0, ChildObjectA.objects.contains, self.existing)
```

Comment:
This assertions are mixed and it's not clear why the number of queries is `1` for `ObjectA` and `0` for `ChildObjectA`. Please check [an example](https://github.com/django/django/pull/13038#discussion_r448849575) in my previous comment and group them.